pkg/adapter/mtping/runner.go

```diff
@@ -101,14 +101,33 @@ func (a *cronJobsRunner) AddSchedule(source *sourcesv1.PingSource) cron.EntryID
 	ctx = observability.WithSpanData(ctx, spanName, int(trace.SpanKindProducer),
 		observability.K8sAttributes(source.Name, source.Namespace, sourcesv1.Resource("pingsource").String()))
 
-	schedule := source.Spec.Schedule
-	if source.Spec.Timezone != "" {
-		schedule = "CRON_TZ=" + source.Spec.Timezone + " " + schedule
-	}
-
 	ctx = kncloudevents.ContextWithMetricTag(ctx, metricTag)
-	id, _ := a.cron.AddFunc(schedule, a.cronTick(ctx, event))
-	return id
+	// If Date is not set, we use Schedule to trigger cron job.
+	if source.Spec.Date == "" {
```

Comment:
Right now if both date and schedule are set, it will respect the date and not schedule. To make it truly mutually exclusive, maybe add an extra validation like [here](https://github.com/knative/eventing/blob/main/pkg/apis/sources/v1/ping_validation.go#L66-L67)

Reply:
Thanks for your review:) Nice comment!